executor/poolmgr/gp.go

```diff
@@ -297,15 +297,29 @@ func (gp *GenericPool) getFetcherUrl(podIP string) string {
 	return fmt.Sprintf("http://%v:8000/", podIP)
 }
 
+func IsIPv6(podIP string) bool {
+	ip := net.ParseIP(podIP)
+	return ip != nil && strings.Contains(podIP, ":")
+}
+
 func (gp *GenericPool) getSpecializeUrl(podIP string, version int) string {
 	u := os.Getenv("TEST_SPECIALIZE_URL")
+	isv6 := IsIPv6(podIP)
+	var baseUrl string
 	if len(u) != 0 {
 		return u
 	}
+	if isv6 == false {
+		baseUrl := fmt.Sprintf("http://%v:8888", podIP)
```

Comment:
subtle thing, but this should be `baseUrl = ...` not `baseUrl := ...`. With the `:=` operator you're creating a new variable inside this scope and hiding the baseUrl variable defined on line 308.